front/src/applications/operationalStudies/helpers/upsertNewProjectedTrains.ts

```diff
@@ -0,0 +1,39 @@
+import type { TrainSpaceTimeData } from 'applications/operationalStudies/types';
+import { type ProjectPathTrainResult, type TrainScheduleResult } from 'common/api/osrdEditoastApi';
+import { convertIsoUtcToLocalTime } from 'utils/date';
+import { deepCloneMap } from 'utils/types';
+
+// TODO: feed the spaceTimeChart with date instead of formatted string
+const formatDatetimeForSpaceTimeChart = (departureTime: string) =>
+  `${convertIsoUtcToLocalTime(departureTime).slice(0, -6)}Z`;
+
+const upsertNewProjectedTrains = (
+  projectedTrains: Map<number, TrainSpaceTimeData>,
+  projectedTrainsToUpsert: Record<string, ProjectPathTrainResult>,
+  trainSchedulesById: Map<number, TrainScheduleResult>
+) => {
+  const newProjectedTrains = deepCloneMap(projectedTrains);
```

Comment:
It should be unnecessary to deep-clone the whole map here. A simple clone (`new Map(projectedTrains)`) should be enough since we copy any projected train we update.